File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{

    string text = get_string("Enter text here: ");

// main for loop
    int lettercount = 0;
    float words = 0;
    int sentence = 0;
    string result = NULL;
    for (int loop = 0, max = strlen(text); loop < max; loop++)
    {
        if ((text[loop] >= 'a' && text[loop] <= 'z') || (text[loop] >= 'A' && text[loop] <= 'Z'))
        {
            lettercount++;
        }

        if (text[loop] == ' ')
        {
            words++;
        }

        if (text[loop] == '.' || text[loop] == '!' || text[loop] == '?')
        {
            sentence++;
        }
    }
/*    printf("Letters: %i \n", lettercount);
    printf("Words: %f \n", words+1); //Added 1 because last word don't end at space character.
    printf("Sentences: %i \n",sentence);
*/

    float L = ( lettercount / (words + 1 ) )*100;
    float S = ( sentence / (words + 1 ) )*100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);
//    printf("L: %f     S: %f",L,S);

    if (grade >= 16)
    {

        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {

        printf("Grade Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n",grade);
    }
}

File: readability.py
import cs50

text = cs50.get_string("Text: ")
alphabets = 0
words = 0
sentences = 0
for eachletter in text:
    if eachletter.isalpha():
        # print(f"Alphabet: {eachletter}")
        alphabets += 1
    elif eachletter == " ":
        # print("WhiteSpace",eachletter,"<-")
        words += 1
    elif eachletter in ".!?":
        sentences += 1
        # print("Sentence ends here")
L = (alphabets / (words + 1)) * 100
S = (sentences / (words + 1)) * 100
grade = int(round(0.0588 * L - 0.296 * S - 15.8))
if (grade >= 16):
    print("Grade 16+");
elif (grade < 1):

    print("Grade Before Grade 1");

else:
    print("Grade {0}".format(grade));
